packages/mui-material-next/src/styles/styleFunctionSx.ts

```diff
@@ -0,0 +1,102 @@
+import {
+  Interpolation,
+  unstable_createStyleFunctionSx,
+  compose,
+  style,
+  display,
+  flexbox,
+  grid,
+  positions,
+  sizing,
+  spacing,
+  border,
+  borderTop,
+  borderRight,
+  borderBottom,
+  borderLeft,
+  borderColor,
+  borderTopColor,
+  borderRightColor,
+  borderBottomColor,
+  borderLeftColor,
+  getValue,
+  createUnaryUnit,
+  handleBreakpoints,
+  responsivePropType,
+  typography,
+} from '@mui/system';
+import { SxProps, Theme } from './Theme.types';
+
+// Palette values should reference the color tokens
+export const color = style({
+  prop: 'color',
+  themeKey: 'sys.color',
+});
+
+export const bgcolor = style({
+  prop: 'bgcolor',
+  cssProperty: 'backgroundColor',
+  themeKey: 'sys.color',
+});
+
+export const backgroundColor = style({
+  prop: 'backgroundColor',
+  themeKey: 'sys.color',
+});
```

Comment:
Not sure how this would work given that the color structure has 2 nodes.

- `sys.color.*`
- `ref.palette.*`

Reply:
I've decided for now to just consider `sys.color`, but we could easily create a custom system fn to support both. It would be even easier if we decide to go with: https://github.com/mui/material-ui/pull/35150. Actually, let me update this fn to support both before we merge this, to avoid surprises. I would pick values from `sys.color` first and them if they don't exist from `ref.palette`.

Reply:
Updated to add support for both palette structures in https://github.com/mui/material-ui/pull/34650/commits/2812dea9cddda758c2b2a28df030ef5fafac73e0